pkg/tbtc/chain_test.go

```diff
@@ -14,32 +16,40 @@ import (
 	"golang.org/x/crypto/sha3"
 )
 
+const localChainOperatorID = chain.OperatorID(1)
+
 var errNilDKGResult = fmt.Errorf("nil DKG result")
 
 type localChain struct {
 	dkgResultSubmissionHandlersMutex sync.Mutex
 	dkgResultSubmissionHandlers      map[int]func(submission *DKGResultSubmittedEvent)
 
-	resultSubmissionMutex sync.Mutex
-	activeWallet          []byte
-	resultSubmitterIndex  group.MemberIndex
+	dkgResultApprovalHandlersMutex sync.Mutex
+	dkgResultApprovalHandlers      map[int]func(submission *DKGResultApprovedEvent)
+
+	dkgResultChallengeHandlersMutex sync.Mutex
+	dkgResultChallengeHandlers      map[int]func(submission *DKGResultChallengedEvent)
+
+	dkgMutex       sync.Mutex
+	dkgState       DKGState
+	dkgResult      *DKGChainResult
+	dkgResultValid bool
 
 	blockCounter       chain.BlockCounter
 	chainConfig        *ChainConfig
 	operatorPrivateKey *operator.PrivateKey
+
+	operatorsIDs map[chain.Address]chain.OperatorID
```

Comment:
Do we need this map? I think the only place where it is used is `GetOperatorID` and we could return the hardcoded `localChainOperatorID` there. Plus, save the address in `localChain` and compare in `GetOperatorID` if the one from the parameter is as expected.

My point here is that the `localChain` implementation allows for only one address and one operator ID while this mapping suggests otherwise. I was even wondering why we don't have a mutex protecting it.

Reply:
See d78aa84f